launcher/lobby/lobby_moc.cpp

```diff
@@ -456,7 +516,31 @@ void Lobby::on_kickButton_clicked()
 
 void Lobby::on_buttonResolve_clicked()
 {
-	//TODO: auto-resolve mods conflicts
+	QStringList toEnableList, toDisableList;
+	for(auto * item : ui->modsList->selectedItems())
+	{
+		auto data = item->data(Qt::UserRole);
```

Comment:
I don't understand why this is something incorrect, from documentation:
Qt::UserRole | 0x0100 | The **first** role that can be used for application-specific purposes.

Also I saw it in multiple examples as correct way

Reply:
I do agree with kambala here.
It is not clear what this user data represents.
Add enum/const with constant for your data

This will allow:
- easy search for all usages of this type of user data
- single place to add new user data / change index of existing
- well-defined place to describe its format/meaning. Right now I need to search for every other access to Qt::UserRole, potentially skip those that refer to other locations, find one near which you've added description about format and scroll a bit up to actually find one

Reply:
Whether your define should be "UserRole" or "UserRole + 1" is a minor point IMO, as long as Qt permits it, but I do agree that this place should have own role

Reply:
Actually you can create an enum starting with Qt::UserRole, and the following roles will be enumerated automatically.

A snippet from my actual production code::
```
        enum ProductRoles
        {
            NameRole = Qt::UserRole + 1,
            PriceRole,
            QuantityRole,
            ImageNameRole
        };
```

And such enum is created for every model separately.

Reply:
> A snippet from my actual production code

Yep, that's what I also always use